### PR TITLE
Add setting to toggle automatic distribution of sidebar boxes

### PR DESCRIPTION
**Pull request by timoludwig**
### Short description

Make the automatic distribution of sidebar boxes opt-in

### Proposed changes

- Improve error messages of view status code tests
- Refactor user settings view
- Add tests for user settings view
- Add setting to toggle automatic distribution of sidebar boxes

### Resolved issues

Fixes: #2033

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/migrations/0060_user_distribute_sidebar_boxes.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        ("cms", "0059_add_poicategory_icon_color"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="user",
+            name="distribute_sidebar_boxes",
+            field=models.BooleanField(
+                default=False,
+                help_text="Enable this option to automatically distribute the sidebar boxes of content forms on screen resolutions wider than 2100px to make ideal use of the space on the screen.",
```
Comment (JoeyStk) [inline integreat_cms/cms/migrations/0060_user_distribute_sidebar_boxes.py:21]: Maybe this was a mistake of mine during the implementation, but actually this also works for Tablet sized screens, that have two sidebar columns beneath the editor. 
grafik
This goes for screen with a width between 768px (md) and 1600px (3xl)

**Review by JoeyStk — APPROVED** [review]
Thank you a lot for this quick fix! This unblocks us from releasing a new version :pray: 
I have one minor note

File: integreat_cms/cms/migrations/0060_user_distribute_sidebar_boxes.py (hunk 1 above)
Comment (timoludwig) [reply integreat_cms/cms/migrations/0060_user_distribute_sidebar_boxes.py]: Good point! :+1:
I now adapted the wording.

**Review by ulliholtgrave — APPROVED** [review]
Thanks. This looks very clean and I like the generic approach for the UserPreferences :+1:

**Pull request merged**